h/schemas/oauth.py

```diff
@@ -12,39 +12,44 @@
 class OAuthCallbackSchema(JSONSchema):
     schema: ClassVar = {  # type: ignore[misc]
         "type": "object",
-        "required": ["code"],
+        "required": ["code", "state"],
         "properties": {
             "code": {"type": "string"},
             "state": {"type": "string"},
-            "error": {"type": "string"},
-            "error_description": {"type": "string"},
         },
     }
 
 
 class OAuthCallbackData(TypedDict):
     code: str
-    state: str | None
-    error: str | None
-    error_description: str | None
```

Comment:
`state` is now required.

`error` and `error_description` were unused and the way that they were implemented here was invalid anyway, see commit message.